### PR TITLE
xtensa: Implement hostfs using simcall

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: arch/xtensa/src/common/xtensa_hostfs.c [hunk 1]
```diff
@@ -0,0 +1,231 @@
+/****************************************************************************
+ * arch/xtensa/src/common/up_hostfs.c
```
Comment (yamt) [inline arch/xtensa/src/common/xtensa_hostfs.c]: oops i should update filename

File: arch/xtensa/src/common/xtensa_hostfs.c [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+ *
+ *   Copyright (C) 2018 Pinecone Inc. All rights reserved.
+ *   Author: Xiang Xiao <xiaoxiang@pinecone.net>
```
Comment (yamt) [inline arch/xtensa/src/common/xtensa_hostfs.c]: @xiaoxiang781216 
is this you?
this file is based on arch/arm/src/common/up_hostfs.c.
may i change this license to apache 2.0?

Comment (xiaoxiang781216) [reply arch/xtensa/src/common/xtensa_hostfs.c]: Sure.

Comment (yamt) [reply arch/xtensa/src/common/xtensa_hostfs.c]: thank you. done.

**Pull request merged**